dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -209,14 +243,101 @@
+		<ItemGroup>
+			<!-- Convert RuntimeIdentifiers (a property) to an item group -->
+			<_RuntimeIdentifiersAsItems Include="$(RuntimeIdentifiers)" Condition=" '$(RuntimeIdentifiers)' != '' " />
+			<_RuntimeIdentifiersAsItems Update="@(_RuntimeIdentifiersAsItems)">
+				<!-- Platform = "rid-arch".Substring (0, "rid-arch".IndexOf ('-')) -->
+				<Platform>$([System.String]::new('%(Identity)').Substring(0, $([System.String]::new('%(Identity)').IndexOf ('-'))))</Platform>
+			</_RuntimeIdentifiersAsItems>
+
+			<!-- Verify that we're not mixing simulator + device RIDs in a single build -->
+			<_RuntimeIdentifierPlatforms Include="@(_RuntimeIdentifiersAsItems -> '%(Platform)')" />
+			<_RuntimeIdentifierDistinctPlatforms Include="@(_RuntimeIdentifierPlatforms->Distinct())" />
+		</ItemGroup>
+
+		<!-- Show an error if we're mixing simulator + device RIDs in a single build -->
+		<Error Condition="@(_RuntimeIdentifierDistinctPlatforms->Count()) > 1" Text="Building for all the runtime identifiers '$(RuntimeIdentifiers)' at the same time isn't possible, because they represent different platform variations." />
+
+		<!-- Execute the inner builds -->
+		<MSBuild
+			Projects="$(MSBuildProjectFile)"
+			Targets="_BuildRidSpecificAppBundle"
+			Properties="RuntimeIdentifier=%(_RuntimeIdentifiersAsItems.Identity);RuntimeIdentifiers=;_IsMultiRidBuild=true">
+			<Output TaskParameter="TargetOutputs" ItemName="_AssemblyPublishDirectories" />
+		</MSBuild>
+
+		<!-- Set the RID-specific subdirectory as metadata -->
+		<ItemGroup>
+			<_AssemblyPublishDirectories>
+				<SpecificSubDirectory>.xamarin/%(RuntimeIdentifier)</SpecificSubDirectory>
+			</_AssemblyPublishDirectories>
+			<_AssemblyPublishInputs Include="@(_AssemblyPublishDirectories -> '%(Identity)/**')" />
+		</ItemGroup>
+	</Target>
+
+	<Target Name="_CreateMergedAppBundle"
+		Inputs="@(_AssemblyPublishInputs)"
+		Outputs="$(DeviceSpecificIntermediateOutputPath)$(_AppBundleName).mergestamp"
+		DependsOnTargets="_RunRidSpecificBuild"
+		>
+		<ItemGroup>
+			<!-- If we're creating a runtimeconfig.json file, then the resulting binary runtimeconfig.bin file can be put in the architecture-specific subdirectory -->
+			<_ArchitectureSpecificFiles Include="$(_RuntimeConfigurationFile)" Condition="'$(GenerateRuntimeConfigurationFiles)' != ''" />
+			<!-- The globalization data file can be put in the architecture-specific subdirectory -->
+			<_ArchitectureSpecificFiles Include="$(_GlobalizationDataFile)" Condition="'$(_GlobalizationDataFile)' != ''" />
+		</ItemGroup>
+
+		<MergeAppBundles
+			SessionId="$(BuildSessionId)"
+			Condition="'$(IsMacEnabled)' == 'true'"
+			ArchitectureSpecificFiles="@(_ArchitectureSpecificFiles)"
+			IgnoreFiles="Info.plist"
+			InputAppBundles="@(_AssemblyPublishDirectories)"
+			OutputAppBundle="$(MSBuildProjectDirectory)/$(_AppBundlePath)"
```

Comment:
And we could potentially have another problem here.

Reply:
@emaf same, I've changed this to be a relative path now.